lib/vehicle_factory.rb

```diff
@@ -0,0 +1,95 @@
+class VehicleFactory
+  attr_reader :created_vehicles
+
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles(vehicles)
+    if vehicles[0][:electric_vehicle_type]
+      vehicles.each do |vehicle_data|
+        vehicle = Vehicle.new({
+          vin: vehicle_data[:vin_1_10],
+          year: vehicle_data[:model_year],
+          make: vehicle_data[:make],
+          model: vehicle_data[:model],
+          engine: :ev,
+          county: vehicle_data[:county]
+          })
+          @created_vehicles << vehicle
+      end
+      @created_vehicles
+    elsif vehicles[0][:record_type]
+      vehicles.each do |vehicle_data|
+        vehicle = Vehicle.new({
+          vin: vehicle_data[:vin],
+          year: vehicle_data[:model_year],
+          make: vehicle_data[:make],
+          model: vehicle_data[:body_type],
+          engine: vehicle_data[:fuel_type],
+          county: vehicle_data[:county]
+          })
+          @created_vehicles << vehicle
+      end
+      @created_vehicles
+    else
+      "Invalid argument, please pass in data from ny or wa_ev registrations" 
+    end
+  end
+
+  def most_popular_make_model
+    make_model_counts = Hash.new(0)
+    @created_vehicles.each do |vehicle|
+      make_model =  "#{vehicle.make} #{vehicle.model}" 
+      make_model_counts[make_model] += 1
+    end
+
+    max_count = 0
+    most_popular_make_model = "No vehicles."
+    make_model_counts.each do |make_model, count|
+      if count > max_count
+        max_count = count
+        most_popular_make_model = make_model
+      end
+    end
+    most_popular_make_model
+  end
+
+  def models_in_a_year(model, year)
+    count = 0
+    @created_vehicles.each do |vehicle|
```

Comment:
There's also a `count` enumerable that could probably help shorten this code too, but same logic! Great job.